shopinvader_api_payment_cart/routers/cart.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2024 ACSONE SA (https://acsone.eu).
+# @author Stéphane Bidoul <stephane.bidoul@acsone.eu>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from typing import Annotated
+
+from fastapi import Depends
+
+from odoo import api
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+)
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.shopinvader_api_cart.routers import cart_router
+from odoo.addons.shopinvader_api_payment.routers.utils import Payable
+from odoo.addons.shopinvader_api_payment.schemas import PaymentData
+
+
+@cart_router.get("/current/payable")
+@cart_router.get("/{uuid}/payable")
+def init(
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated["ResPartner", Depends(authenticated_partner)],
+    uuid: str | None = None,
+) -> PaymentData | None:
+    """Prepare payment data for the current cart.
+
+    This route is authenticated, so we can verify the cart
+    is accessible by the authenticated partner.
+    """
+    cart = env["sale.order"]._find_open_cart(partner.id, uuid)
+    if not cart:
+        return None
+
+    sale_order = env["sale.order"].sudo().browse(cart.id)
+    payment_data = {
+        "payable": Payable(
+            payable_id=cart.id,
+            payable_model="sale.order",
+            payable_reference=sale_order.name,
+            amount=sale_order.amount_total,
+            currency_id=sale_order.currency_id.id,
+            partner_id=sale_order.partner_id.id,
+            company_id=sale_order.company_id.id,
+        ).model_dump_json(),
+        "payable_reference": sale_order.name,
+        "amount": sale_order.amount_total,
+        "currency_code": sale_order.currency_id.name,
```

Comment:
Might be nice to return `amount_formatted` = `sale_order.curency_id.format(amount)`.
To simplify the life of the front end dev, and avoid accidentally displaying 17.40000000000000001 as amount to pay.

Reply:
Done